File: Creational/FactoryMethod/Java/Structure/ConcreteProductB.java
/**
 * ConcreteProductB - implementation cụ thể khác của Product
 */
public class ConcreteProductB implements Product {
    
    @Override
    public void operation() {
        System.out.println("ConcreteProductB đang thực hiện thao tác");
    }
    
    @Override
    public String getName() {
        return "Sản phẩm B";
    }
}

File: Creational/FactoryMethod/TypeScript/Structure/ConcreteProductB.ts
import { Product } from "./Product";

/**
 * ConcreteProductB - Concrete implementation of Product
 * Represents a specific type of product created by ConcreteCreatorB
 */
export class ConcreteProductB extends Product {
  operation(): string {
    return "ConcreteProductB operation";
  }
}
